catalog/core/catalog-core-migratable/src/test/java/org/codice/ddf/catalog/migratable/impl/MigrationFileWriterTest.java

```diff
@@ -0,0 +1,179 @@
+import java.io.IOException;
+import java.io.ObjectOutputStream;
+import java.io.OutputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.codice.ddf.migration.MigrationException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import ddf.catalog.data.Result;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.data.impl.ResultImpl;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({MigrationFileWriter.class, File.class})
+public class MigrationFileWriterTest {
+
+    private static final int RESULT_COUNT = 5;
+
+    private static final Path DDF_BASE_PATH = Paths.get("ddf");
+
+    @Mock
+    private Path mockPath;
+
+    @Mock
+    private File mockFile;
+
+    @Mock
+    private FileOutputStream mockFileOutputStream;
+
+    @Mock
+    private BufferedOutputStream mockBufferedOutputStream;
+
+    @Mock
+    private ObjectOutputStream mockObjectOutputStream;
+
+    private MigrationFileWriter fileWriter;
+
+    @Before
+    public void setup() throws Exception {
+        fileWriter = new MigrationFileWriter() {
+            @Override
+            ObjectOutputStream createObjectStream(OutputStream source) {
+                return mockObjectOutputStream;
+            }
+
+            @Override
+            BufferedOutputStream createBufferedStream(OutputStream source) {
+                return mockBufferedOutputStream;
+            }
+
+            @Override
+            FileOutputStream createFileStream(File file) {
+                return mockFileOutputStream;
+            }
+        };
+
+        when(mockPath.toFile()).thenReturn(mockFile);
+    }
+
+    private List<Result> loadList() {
+        final List<Result> results = new ArrayList<>();
+        for (int i = 0; i < RESULT_COUNT; i++) {
+            results.add(new ResultImpl(new MetacardImpl()));
+        }
+        return results;
+    }
+
+    @Test
+    public void testInitCallsMkDir() throws Exception {
+        when(mockFile.exists()).thenReturn(false);
+        when(mockFile.mkdir()).thenReturn(true);
+
+        fileWriter.init(mockPath);
+
+        verify(mockFile).mkdir();
+    }
+
+    @Test(expected = MigrationException.class)
+    public void testInitFailsToCreateDirectory() throws Exception {
+
+        when(mockFile.exists()).thenReturn(false);
+        when(mockFile.mkdir()).thenReturn(false);
+
+        fileWriter.init(mockPath);
+    }
+
+    @Test
+    public void testWriteSuccess() throws Exception {
+        final List<Result> results = loadList();
+        when(mockFile.exists()).thenReturn(true);
+
+        fileWriter.writeMetacards(mockFile, results);
+
+        verify(mockObjectOutputStream, times(RESULT_COUNT)).writeObject(anyObject());
```

Comment:
Can't do that - in the implementation we are passing the metacard into the ctor of a MetacardImpl so that we can standardize the fields into real fields based on the current DDF instance. This is necessary when other metacard implementations enter the picture. How on earth do we mock that since the addresses are identical but not registered as the same?